packages/consumption/src/modules/attributes/local/LocalAttributeDeletionStatus.ts

```diff
@@ -0,0 +1,33 @@
+import { serialize, validate } from "@js-soft/ts-serval";
+import { CoreDate, CoreSerializable, ICoreDate } from "@nmshd/transport";
+
+// TODO: check all places that need to be adjusted adding deletionStatus to LocalAttribute
+export interface LocalAttributeDeletionStatusJSON {
+    toBeDeletedAt?: string;
+    toBeDeletedByPeerAt?: string;
+    deletedByPeer?: string;
+}
+
+export interface ILocalAttributeDeletionStatus {
+    toBeDeletedAt?: ICoreDate;
+    toBeDeletedByPeerAt?: ICoreDate;
+    deletedByPeer?: ICoreDate;
+}
```

Comment:
Use an enum instead, as discussed in the meeting.

Reply:
looks great.

Reply:
We'll have to check, if indeed no two status can be set at the same time. E.g. if the owner sends a DeleteAttributeRequest and the peer accepts, `toBeDeleted` will be set. If, however, in the meantime the owner deletes the Attribute and sends a Notification, which would set `deletedByPeer` for the peer, we cannot use an enum. We should discuss this in the Team Sync today, when talking about the other Attribute Deletion Use Cases.

Reply:
We'll need a unique enum item to represent every possible deletion state either way. I'm not fond of the idea of having a two-dimensional deletion state, no reason to do that. With respect to your example, if you don't want to throw away any information, just add another "compound" state `toBeDeletedAndDeletedByPeer` to the enum.

Whether the peer needs to know that the attribute has been deleted by the owner in the meanwhile, i.e. whether we need that extra compound state is a separate question though. We'll discuss this in the upcoming meeting 👍